go/proxyd/server.go

```diff
@@ -274,6 +253,107 @@ func (s *Server) handleSingleRPC(ctx context.Context, req *RPCReq) (*RPCRes, boo
 	return backendRes, false
 }
 
+func (s *Server) handleBatchRPC(ctx context.Context, reqs []json.RawMessage) ([]*RPCRes, bool) {
```

Comment:
we could also have RPC requests that aren't batched also using this.

Reply:
I kind of like the idea of treating everything beyond the server code as a batch request - i.e. we'd do the conversion of a single request into a batch in the server component, then pass that batch to the `Backend`.